<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1544</width>
    <height>627</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="preorderButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>430</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Прямой обход</string>
    </property>
   </widget>
   <widget class="QPushButton" name="inorderButton">
    <property name="geometry">
     <rect>
      <x>169</x>
      <y>430</y>
      <width>181</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Обход в порядке
возрастания ключа</string>
    </property>
   </widget>
   <widget class="QPushButton" name="postorderButton">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>430</y>
      <width>161</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Обратный обход</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>200</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите ключ</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="resultTextEdit">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>280</y>
      <width>451</width>
      <height>141</height>
     </rect>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>20</y>
      <width>411</width>
      <height>441</height>
     </rect>
    </property>
    <property name="animated">
     <bool>true</bool>
    </property>
    <column>
     <property name="text">
      <string notr="true"/>
     </property>
    </column>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите строку</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>408</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>1</x>
      <y>259</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>460</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>70</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите ключ</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>1</x>
      <y>155</y>
      <width>16</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="stringLineEdit">
    <property name="geometry">
     <rect>
      <x>159</x>
      <y>104</y>
      <width>125</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>480</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить дерево</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit">
    <property name="geometry">
     <rect>
      <x>159</x>
      <y>69</y>
      <width>125</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit1">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>200</y>
      <width>125</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="removeButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>210</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить строку по ключу</string>
    </property>
   </widget>
   <widget class="QPushButton" name="searchButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>160</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Найти строку по ключу</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>70</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Добавить ключ + строку</string>
    </property>
   </widget>
   <widget class="QPushButton" name="balanceButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>490</y>
      <width>221</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Сбалансировать дерево</string>
    </property>
   </widget>
   <widget class="QPushButton" name="taskButton">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>490</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить из дерева ветвь с вершиной, 
имеющей заданный ключ</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit2">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>500</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>1001</x>
      <y>121</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите строку</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addNewButton">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>170</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Добавить ключ + строку
 в новое дерево</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="stringLineEdit3">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>115</y>
      <width>125</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>1001</x>
      <y>81</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите ключ</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit3">
    <property name="geometry">
     <rect>
      <x>1130</x>
      <y>80</y>
      <width>125</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget1">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>240</y>
      <width>281</width>
      <height>211</height>
     </rect>
    </property>
    <property name="animated">
     <bool>true</bool>
    </property>
    <column>
     <property name="text">
      <string notr="true"/>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="addTreeButton">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>460</y>
      <width>131</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Вставить дерево</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeTreeButton">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>460</y>
      <width>131</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Удалить дерево</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>20</y>
      <width>211</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>ОПЕРАЦИИ С ПОДДЕРЕВОМ</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>10</y>
      <width>241</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>ПОИСК НАИМЕНЬШЕГО ОБЩЕГО
        ПРЕДКА ДЛЯ ДВУХ УЗЛОВ</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit4">
    <property name="geometry">
     <rect>
      <x>1439</x>
      <y>99</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>100</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите ключ 1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>1310</x>
      <y>130</y>
      <width>111</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Введите ключ 2</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyLineEdit5">
    <property name="geometry">
     <rect>
      <x>1440</x>
      <y>130</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="resultTextEdit1">
    <property name="geometry">
     <rect>
      <x>1330</x>
      <y>240</y>
      <width>181</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="ancButton">
    <property name="geometry">
     <rect>
      <x>1370</x>
      <y>170</y>
      <width>83</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Найти</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1544</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
